common/utils/src/main/resources/error/error-classes.json

```diff
@@ -133,6 +133,48 @@
       "Could not load Protobuf class with name <protobufClassName>. <explanation>."
     ]
   },
+  "CANNOT_LOAD_STATE_STORE" : {
+    "message" : [
+      "An error occurred during loading state."
+    ],
+    "subClass" : {
+      "CANNOT_READ_CHECKPOINT" : {
+        "message" : [
+          "Cannot read RocksDB checkpoint metadata of version <versionLine>"
+        ]
+      },
+      "CANNOT_READ_DELTA_FILE" : {
+        "message" : [
+          "Error reading delta file <fileToRead> of <clazz>: <message>"
+        ]
+      },
+      "CANNOT_READ_SNAPSHOT_FILE" : {
+        "message" : [
+          "Error reading snapshot file <fileToRead> of <clazz>: <message>"
+        ]
+      },
+      "CANNOT_READ_STREAMING_STATE_FILE" : {
+        "message" : [
+          "Error reading streaming state file of <fileToRead> does not exist. If the stream job is restarted with a new or updated state operation, please create a new checkpoint location or clear the existing checkpoint location."
+        ]
+      },
+      "UNEXPECTED_FILE_SIZE" : {
+        "message" : [
+          "Copied <dfsFile> to <localFile>, expected <expectedSize> bytes, found <localFileSize> bytes."
+        ]
+      },
+      "UNRELEASED_THREAD_ERROR": {
+        "message" : [
+          "<loggingId>: RocksDB instance could not be acquired by <newAcquiredThreadInfo> as it was not released by <acquiredThreadInfo> after <timeWaitedMs> ms.",
+          "Thread holding the lock has trace: <stackTraceOutput>"
+        ]
+      },
+      "WRAPPER" : {
+        "message" : [""]
+      }
+    },
+    "sqlState" : "40000"
```

Comment:
Transaction rollback? Are you sure?

Reply:
Hmm, I wasn't sure what to put. What would you suggest? The CANNOT_READ_... subclasses fall under [CANNOT_READ_FILE sqlState: KD003](https://docs.databricks.com/error-messages/cannot-read-file-error-class.html), but I don't think the others do.

Reply:
Are any of these Internal errors? I.e they should result in support tickets? Those should be taken out.
I find it interesting that you talk about RocksDB, did teh user pick RocksDB or is that an internal service?

Maybe I need a bit more context on what the STATE STORE is...

Reply:
These are not about inputs and outputs from the point of end-to-end query. You can imagine that there is an operator which have to retain accumulators over the streaming query's lifetime (I'm over-simplifying but I guess this is more SQL friendly), and we are checkpointing accumulators for that microbatch to the durable file system.

So it's not directly related to the data users read from/write to, but the internal data maintained by the streaming query.